### PR TITLE
New GUI menu and conversion options

### LINKED ISSUE #408: Add user interface to support Dominion conversion process
After talking with @CalebKleppner we arrived at the following:

The tabulator needs to ingest and process Dominion json data for various elections (Wyoming, Alaska, Kansas Democratic primaries, and eventually NYC local).  

To allow users to input, convert (verification / auditing - #404) and tabulate, we need a way for users to specify Dominion as a data provider.  For Dominion data providers, the user needs to specify an input Dominion folder location.  (First Vote Column Index, etc... are not needed)

**CLI support for conversion** this can be run from CLI: user supplies input Dominion data folder location.  Tabulator will input Dominion cvr json data from this folder.
Tabulator will export generic cvr csv to the same folder (#404) with a file per contest.

**GUI support for conversion**

User can click a "convert dominion cvr json to generic csv" button on the main panel.  This opens a file chooser.  User can select a file to convert.  Exported files will be written to the same folder as the input file.  Per @CalebKleppner no need to select an export folder.

Moved the Tabulation support tasks to a new ticket: #418

### LINKED ISSUE #421: Incorporate `convert-to-cdf` CLI functionality in GUI
Per Slack conversation with @tarheel, this CLI option is different from the `generateCdfJson` checkbox that's currently in the GUI because:

> generateCdfJson generates an output file in CDF format. convert-to-cdf just generates an input CVR file in CDF format. the input and output formats are quite similar; the output just adds info to each cast vote record indicating how it was allocated, and it goes round by round

### PR DESCRIPTION
**Pull request by HEdingfield**
Fixes #408. Fixes #421.

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/GuiConfigController.java [hunk 1]
```diff
@@ -317,6 +317,27 @@ public void menuItemTabulateClicked() {
...
+   * Create and launch ConvertToCdfService from the saved config path.
+   */
+  public void menuItemConvertToCdfClicked() {
+    if (checkForSaveAndExecute()) {
+      if (GuiContext.getInstance().getConfig() != null) {
+        setGuiIsBusy(true);
+        ConvertToCdfService service = new ConvertToCdfService(selectedFile.getAbsolutePath());
+        service.setOnSucceeded(event -> setGuiIsBusy(false));
+        service.setOnCancelled(event -> setGuiIsBusy(false));
+        service.setOnFailed(event -> setGuiIsBusy(false));
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/GuiConfigController.java]: Can we factor this out into a helper? It looks like we're doing it in a few places.

File: src/main/java/network/brightspots/rcv/GuiConfigController.java [hunk 2]
```diff
@@ -952,4 +973,34 @@ protected Void call() {
...
+
+    @Override
+    protected Task<Void> createTask() {
+      Task<Void> task =
+          new Task<>() {
+            @Override
+            protected Void call() {
+              TabulatorSession session = new TabulatorSession(configPath);
+              session.convertToCdf();
+              return null;
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/GuiConfigController.java]: If the return type is Void, why do you have to have an explicit return statement?

File: src/main/java/network/brightspots/rcv/GuiConfigController.java [hunk 3]
```diff
@@ -338,6 +339,27 @@ public void menuItemConvertToCdfClicked() {
...
+    dc.setInitialDirectory(new File(FileUtils.getUserDirectory()));
+    dc.setTitle("Dominion Data Folder");
+    File dominionDataFolderPath = dc.showDialog(GuiContext.getInstance().getMainWindow());
+    if (dominionDataFolderPath != null) {
+      setGuiIsBusy(true);
+      ConvertDominionService service = new ConvertDominionService(
+          dominionDataFolderPath.getAbsolutePath());
+      service.setOnSucceeded(event -> setGuiIsBusy(false));
+      service.setOnCancelled(event -> setGuiIsBusy(false));
+      service.setOnFailed(event -> setGuiIsBusy(false));
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/GuiConfigController.java]: same

File: src/main/java/network/brightspots/rcv/GuiConfigController.java (hunk 2 above)
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/GuiConfigController.java]: Pretty sure it was boilerplate for thread management. [This](https://stackoverflow.com/a/15425295/3846321) might shed some light on that.

File: src/main/java/network/brightspots/rcv/GuiConfigController.java (hunk 1 above)
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/GuiConfigController.java]: Done.

File: src/main/java/network/brightspots/rcv/GuiConfigController.java (hunk 3 above)
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/GuiConfigController.java]: Done.

**Comment by HEdingfield** [comment]
Filed NPE change in #431.

**Pull request merged**